ros2doctor/ros2doctor/verb/call.py

```diff
@@ -0,0 +1,178 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import socket
+import struct
+import threading
+import time
+
+import rclpy
+from rclpy.executors import MultiThreadedExecutor
+from rclpy.node import Node
+from ros2doctor.verb import VerbExtension
+
+from std_msgs.msg import String
+
+DEFAULT_GROUP = '225.0.0.1'
+DEFAULT_PORT = 49150
+DEFAULT_TOPIC = '/knockknock'
+summary_table = {}
+
+
+class CallVerb(VerbExtension):
+    """Publish and subscribe, multicast send and receive, print table."""
+
+    def main(self, *, args):
+        rclpy.init()
+        pub_node = Talker()
+        sub_node = Listener()
+
+        executor = MultiThreadedExecutor()
+        executor.add_node(pub_node)
+        executor.add_node(sub_node)
+        try:
+            count = 0
+            _spawn_summary_table()
+            while True:
+                if (count % 20 == 0 and count != 0):
+                    format_print(summary_table)
+                    _spawn_summary_table()
+                # pub/sub threads
+                executor.spin_once()
+                executor.spin_once()
```

Comment:
This is tricky, since it doesn't guarantee that one iteration of this loop will result exactly one timer callback and one subscription callback. It could be two timer callbacks or two subscription callbacks, depending on the queue of events.

It may make sense to run the executor in it's own thread and just call `spin()`.

Reply:
> It may make sense to run the executor in it's own thread and just call `spin()`.

Do you mean like `threading.Thread(target=executor.spin)`? I thought `executor` already has multiple threads running, wouldn't putting the executor in another thread be repetitive?

Reply:
> Do you mean like threading.Thread(target=executor.spin)? 

Yes.

The MultiThreadedExecutor does use multiple threads to execute callbacks, but it still needs someone to call `spin_once` (or `spin`) to poll for callbacks to submit for execution. Versus the SingleThreadedExecutor, which executes the callbacks in the same thread that `spin` is called in.

I'm not sure if we really need to use a MultiThreadedExecutor in this case; I think either will executor will work.